go/tasks/plugins/k8s/dask/dask.go

```diff
@@ -30,68 +30,72 @@ const (
+}
 
-	var defaultEnvVars []v1.EnvVar
-	if taskTemplate.GetContainer().GetEnv() != nil {
-		for _, keyValuePair := range taskTemplate.GetContainer().GetEnv() {
-			defaultEnvVars = append(defaultEnvVars, v1.EnvVar{Name: keyValuePair.Key, Value: keyValuePair.Value})
+func replacePrimaryContainer(spec *v1.PodSpec, primaryContainerName string, container v1.Container) error {
+	for i, c := range spec.Containers {
+		if c.Name == primaryContainerName {
+			spec.Containers[i] = container
+			return nil
 		}
 	}
+	return errors.Errorf(errors.BadTaskSpecification, "primary container [%v] not found in pod spec", primaryContainerName)
+}
 
-	containerResources, err := flytek8s.ToK8sResourceRequirements(defaultContainerSpec.GetResources())
-	if err != nil {
-		return nil, err
+func removeInterruptibleConfig(spec *v1.PodSpec, taskCtx pluginsCore.TaskExecutionContext) {
+	if !taskCtx.TaskExecutionMetadata().IsInterruptible() {
+		return
 	}
 
-	jobRunnerContainer := v1.Container{
-		Name:      "job-runner",
-		Image:     defaultImage,
-		Args:      defaultContainerSpec.GetArgs(),
-		Env:       defaultEnvVars,
-		Resources: *containerResources,
+	// Tolerations
+	interruptlibleTolerations := config.GetK8sPluginConfig().InterruptibleTolerations
+	newTolerations := []v1.Toleration{}
+	for _, toleration := range spec.Tolerations {
+		if !slices.Contains(interruptlibleTolerations, toleration) {
+			newTolerations = append(newTolerations, toleration)
+		}
 	}
+	spec.Tolerations = newTolerations
 
-	templateParameters := template.Parameters{
-		TaskExecMetadata: taskCtx.TaskExecutionMetadata(),
-		Inputs:           taskCtx.InputReader(),
-		OutputPath:       taskCtx.OutputWriter(),
-		Task:             taskCtx.TaskReader(),
+	// Node selectors
+	interruptibleNodeSelector := config.GetK8sPluginConfig().InterruptibleNodeSelector
+	for key := range spec.NodeSelector {
+		if _, ok := interruptibleNodeSelector[key]; ok {
+			delete(spec.NodeSelector, key)
+		}
 	}
-	if err = flytek8s.AddFlyteCustomizationsToContainer(ctx, templateParameters,
-		flytek8s.ResourceCustomizationModeMergeExistingResources, &jobRunnerContainer); err != nil {
 
-		return nil, err
+	// Node selector requirements
+	interruptibleNodeSelectorRequirements := config.GetK8sPluginConfig().InterruptibleNodeSelectorRequirement
+	nodeSelectorTerms := spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
+	for i := range nodeSelectorTerms {
+		nst := &nodeSelectorTerms[i]
+		matchExpressions := nst.MatchExpressions
+		newMatchExpressions := []v1.NodeSelectorRequirement{}
+		for _, matchExpression := range matchExpressions {
+			if !nodeSelectorRequirementsAreEqual(matchExpression, *interruptibleNodeSelectorRequirements) {
+				newMatchExpressions = append(newMatchExpressions, matchExpression)
+			}
+		}
+		nst.MatchExpressions = newMatchExpressions
 	}
+}
```

Comment:
An alternative solution would be to create a wrapper over the `TaskExecutionContext` / `TaskExecutionMetadata` which forces`interruptible` to be false. This would stop them from ever being injected without adding more parameters. Something like:
```
struct daskTaskExecutionContext {
    taskExecutionContext
    metadata TaskExecutionMetadata
}

func (d *daskTaskExecutionContext) GetMetadata() TaskExecutionMetadata{
    return metadata
}

type daskTaskExecutionMetadata {
    taskExecutionMetadata
}

func (d *daskTaskExecutionMetadata) IsInterruptible() bool {
    return falsee
}
```
the function and type names are all wrong (don't know off the top of my head), but basically just forcing interruptible to always be false for dask tasks.

Reply:
That sounds good to me - I'll try to implement that!